src/PrestaShopBundle/Resources/views/Admin/Sell/Catalog/Product/Form/attachments_form_theme.html.twig

```diff
@@ -0,0 +1,92 @@
+{# **
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * #}
+{% extends 'PrestaShopBundle:Admin/TwigTemplateForm:prestashop_ui_kit_base.html.twig' %}
+
+{% block collection_widget %}
+    {%- block attached_file_row -%}
+      <tr class="attached_file_row" id="attached_file_row_{{ form.vars.name }}">
+        {% for child in form.children %}
```

Comment:
With the comment I suggested earlier about adding a new `PreviewType` this should be improvable a lot

Once the preview type can handle itself we can use a `{{- block('form_rows') -}}` to replace this big loop (because that's basically what the `form_rows` block does

Reply:
Once `form_rows` is used, you can just override `form_row` inside this sub block this should only customize the row for these children( not all of them), And then you can hide the label and only use `form_widget` and `form_errors`
This would make the whole theme shorter and easier to read

Reply:
I don't think so. previewType will not be in <td>, so we stiill need to render each of them manually to add it

Reply:
yeah so i tried adding option in recently introduced PreviewType to render it in `td` but then it also requires addding option to `IconButtonType` so I think it just means we need to add the `td` manually instead of adapting every component template?